src/ado2gh/Commands/WaitForMigrationCommand.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using OctoshiftCLI.Extensions;
+
+[assembly: InternalsVisibleTo("OctoshiftCLI.Tests")]
+namespace OctoshiftCLI.AdoToGithub.Commands
+{
+    public class WaitForMigrationCommand : Command
+    {
+        internal int WaitIntervalInSeconds = 10;
+
+        private readonly OctoLogger _log;
+        private readonly GithubApiFactory _githubApiFactory;
+
+        public WaitForMigrationCommand(OctoLogger log, GithubApiFactory githubApiFactory) : base("wait-for-migration")
+        {
+            _log = log;
+            _githubApiFactory = githubApiFactory;
+
+            Description = "Waits for migration(s) to finish and reports all in progress and queued ones.";
+            Description += Environment.NewLine;
+            Description += "Note: Expects GH_PAT env variables to be set.";
+
+            var githubOrg = new Option<string>("--github-org") { IsRequired = true };
+            var migrationId = new Option<string>("--migration-id")
+            {
+                IsRequired = false,
+                Description = "Waits for the specified migration to finish."
+            };
+            var verbose = new Option("--verbose") { IsRequired = false };
+
+            AddOption(githubOrg);
+            AddOption(migrationId);
+            AddOption(verbose);
+
+            Handler = CommandHandler.Create<string, string, bool>(Invoke);
+        }
+
+        public async Task Invoke(string githubOrg, string migrationId = null, bool verbose = false)
+        {
+            _log.Verbose = verbose;
+
+            var hasMigrationId = !migrationId.IsNullOrWhiteSpace();
+
+            _log.LogInformation(
+                $"Waiting for {(hasMigrationId ? $"migration {migrationId}" : "all migrations")} to finish...");
+            _log.LogInformation($"GITHUB ORG: {githubOrg}");
+            if (hasMigrationId)
+            {
+                _log.LogInformation($"MIGRATION ID: {migrationId}");
+            }
+
+            var githubApi = _githubApiFactory.Create();
+            var githubOrgId = await githubApi.GetOrganizationId(githubOrg);
+
+            while (true)
+            {
+                if (hasMigrationId)
+                {
+                    var specifiedMigrationState = await githubApi.GetMigrationState(migrationId);
+                    switch (specifiedMigrationState)
+                    {
+                        case RepositoryMigrationStatus.Failed:
+                            {
```

Comment:
are these braces needed?

Reply:
I wonder why didn't `Rider` warn me about it, but they can most certainly be removed 😃 
addressed in dc1a57c0c399b2dc339ff6fcbae5058e0adae071

Reply:
addressed in dc1a57c0c399b2dc339ff6fcbae5058e0adae071